crates/lumina_client/src/ui/game_ui/health.rs

```diff
@@ -10,10 +17,24 @@ impl Plugin for HealthUiPlugin {
         app.register_typst_asset::<GameUi>()
             .compile_typst_func::<GameUi, HealthFunc>()
             .init_resource::<HealthFunc>()
-            .insert_resource(HealthFunc {
-                current_hp: 100.0, // Set initial HP
-                max_hp: 100.0,     // Set max HP
-            });
+            .add_systems(Update, update_health_ui);
+    }
+}
+
+fn update_health_ui(
+    q_spaceships: Query<
+        (&Health, &MaxHealth, &PlayerId),
+        (Changed<Health>, With<Spaceship>, With<SourceEntity>),
+    >,
+    local_player_id: Res<LocalPlayerId>,
+    mut health_func: ResMut<HealthFunc>,
+) {
+    for (health, max_health, player_id) in q_spaceships.iter() {
+        if player_id == &local_player_id.0 {
+            health_func.current_hp = **health as f64;
+            health_func.max_hp = **max_health as f64;
+            break;
+        }
```

Comment:
```suggestion
    for (health, max_health, player_id) in q_spaceships.iter().filter(|id| *id == local_player_id.0) {
          health_func.current_hp = **health as f64;
          health_func.max_hp = **max_health as f64;
     }
```

Might be more elegant this way? haha and also something new for you, u can search up about rust iterators, they are wonderful constructs xd